File: dotnet/Lexy.Poc/Lexy.Poc.Core/Language/ScenarioFunctionName.cs
using System.Collections.Generic;
using Lexy.Poc.Core.Parser;

namespace Lexy.Poc.Core.Language
{
    public class ScenarioFunctionName : Node
    {
        public string Value { get; private set; }

        public INode Parse(IParserContext context)
        {
            var line = context.CurrentLine;
            Value = line.Tokens.TokenValue(1);
            return this;
        }

        public ScenarioFunctionName(SourceReference reference) : base(reference)
        {
        }

        public override string ToString() => Value;

        protected override IEnumerable<INode> GetChildren()
        {
            yield break;
        }

        protected override void Validate(IValidationContext context)
        {
        }
    }
}

File: typescipt/existing/src/language/scenarios/scenarioFunctionName.ts



namespace Lexy.Compiler.Language.Scenarios;

public class ScenarioFunctionName : Node
{
   public string Value { get; private set; }

   public ScenarioFunctionName(SourceReference reference) : base(reference)
   {
   }

   public void Parse(IParseLineContext context)
   {
     var line = context.Line;
     Value = line.Tokens.TokenValue(1);
   }

   public override string ToString()
   {
     return Value;
   }

   public override IEnumerable<INode> GetChildren()
   {
     yield break;
   }

   protected override void Validate(IValidationContext context)
   {
   }

   public bool IsEmpty()
   {
     return string.IsNullOrEmpty(Value);
   }
}
